### PR TITLE
GEOMESA-1685 Fixing NOT ID filters

### PR DESCRIPTION
**Pull request by elahrvivaz**
* Note: requires iterator jars to be updated

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoLazyFilterTransformIterator.scala [hunk 1]
```diff
@@ -78,6 +79,14 @@ class KryoLazyFilterTransformIterator extends
       case (None, Some(s))    => s
       case (Some(c), Some(s)) => (sf) => c.evaluate(sf) && s(sf)
     }
+
+    setId = if (index.serializedWithId || cql.isEmpty) { () => {} } else {
```
Comment (jnh5y) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoLazyFilterTransformIterator.scala]: Ok, why does it matter if the cql filter isn't used?

Comment (jnh5y) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoLazyFilterTransformIterator.scala]: Oh, nevermind.  I'm silly....

File: geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/visitor/IdExtractingVisitor.scala [hunk 2]
```diff
@@ -42,13 +42,10 @@ class IdExtractingVisitor extends FilterVisitor {
     (andOption(yes).orNull, andOption(no).orNull)
   }
 
-  override def visit(f: Not, data: AnyRef): AnyRef = {
-    val (yes, no) = f.getFilter.accept(this, data).asInstanceOf[(Filter, Filter)]
-    (if (yes == null) null else ff.not(yes), if (no == null) null else ff.not(no))
```
Comment (jnh5y) [inline geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/visitor/IdExtractingVisitor.scala]: Ok, this is a little bit mind-bendy for me...  What were we doing here?

Comment (elahrvivaz) [inline geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/visitor/IdExtractingVisitor.scala]: before it was recursing into NOTs and extracting any IDs from there - now if we see a NOT we just stop and return no match. It's possible we might miss some double-negative, but detecting that would be kind of complicated I think.

Comment (jnh5y) [inline geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/visitor/IdExtractingVisitor.scala]: SimplifyingFilterVisitor?  https://github.com/geotools/geotools/blob/master/modules/library/main/src/main/java/org/geotools/filter/visitor/SimplifyingFilterVisitor.java

I thought we were using it...

**Pull request merged**